packages/common/src/HardhatConfig.js

```diff
@@ -27,12 +28,32 @@ function getHardhatConfig(configOverrides, workingDir = "./") {
         },
       },
     },
+    ovm: {
+      solcVersion: solcVersion,
+    },
     networks: {
       hardhat: {
         gas: 11500000,
         blockGasLimit: 11500000,
         allowUnlimitedContractSize: false,
         timeout: 1800000,
+        testBlacklist: ["ovm"],
+      },
+      optimism: {
+        url: "http://127.0.0.1:8545",
+        accounts: { mnemonic: "test test test test test test test test test test test junk" },
+        // This sets the gas price to 0 for all transactions on L2. We do this because account balances are not yet
+        // automatically initiated with an ETH balance.
+        gasPrice: 0,
+        // This sets the network as using the ovm and ensure contract will be compiled against that.
+        ovm: true,
+        // We use custom logic to only compile contracts within the listed directories, as opposed to choosing which
+        // ones to ignore, because there are more contracts to ignore than to include.
+        compileWhitelist: [
```

Comment:
Contracts are limited to those that don't have libraries, which the Optimism-solc compiler doesn't quite handle yet: [issue](https://github.com/ethereum-optimism/solidity/issues/29)